.storybook/test-runner.ts

```diff
@@ -155,25 +157,30 @@ async function expectStoryToMatchSnapshot(
         await Promise.all(waitForSelector.map((selector) => page.waitForSelector(selector)))
     }
 
-    await page.waitForTimeout(400) // Wait for effects to finish
-
-    // Wait for all images to load
-    await page.waitForFunction(() =>
-        Array.from(document.querySelectorAll('img')).every((i: HTMLImageElement) => i.complete)
-    )
-
     // snapshot light theme
     await page.evaluate(() => {
         document.body.setAttribute('theme', 'light')
     })
 
+    // Wait for all images to load
+    await page.waitForFunction(() => Array.from(document.images).every((i: HTMLImageElement) => i.complete))
+    await waitForPageReady(page)
+    await page.waitForLoadState('networkidle')
+    await page.waitForTimeout(2000)
```

Comment:
This is a long wait, it might slow the test suite down. Did you try different values or just this 2sec? Curious if like 500ms does the trick.